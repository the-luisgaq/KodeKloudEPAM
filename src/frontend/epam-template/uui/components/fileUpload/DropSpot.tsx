import * as React from 'react';
import { cx, IHasCX, IHasRawProps } from '@epam/uui-core';
import { DropSpot as UuiDropSpot, DropSpotRenderParams, UploadFileToggler } from '@epam/uui-components';
import { FlexRow, IconContainer } from '../layout';
import { LinkButton } from '../buttons';
import { RichTextView, Text } from '../typography';
import { ReactComponent as ShapeIcon } from '@epam/assets/icons/file-cloud_upload-fill.svg';
import { i18n } from '../../i18n';
import css from './DropSpot.module.scss';

export interface DropSpotProps extends IHasRawProps<React.HTMLAttributes<HTMLDivElement>>, IHasCX {
    /**
    * Defines the infoText. InfoText is additional information that can be rendered at the bottom of the AttachmentArea.
    */
    infoText?: React.ReactNode;
    /**
    * Called when files was added to the DropSpot.
    */
    onUploadFiles(files: File[]): any;
    /**
    * Hint for expected file type in file upload controls. See {@link https://developer.mozilla.org/en-US/docs/web/html/element/input#accept} for details
    */
    accept?: string;
    /**
    * Whether to allow single or multiple values. See {@link https://developer.mozilla.org/en-US/docs/web/html/element/input#multiple} for details
    */
    single?: boolean;
}

export function DropSpot(props: DropSpotProps) {
    const getInfoText = typeof props.infoText === 'string'
        ? (
            <Text lineHeight="24" size="24" fontSize="14" color="tertiary">
                {props.infoText}
            </Text>
        ) : props?.infoText;

    const renderAttachmentArea = (params: DropSpotRenderParams) => {
        return (
            <div className={ cx(css.root, props.cx, params.isDragStart && css.dropStart, params.isDraggingOver && css.dropOver) } { ...props.rawProps }>
                <div { ...params.eventHandlers } className={ css.dropArea }>
                    <FlexRow size="24">
                        <IconContainer icon={ ShapeIcon } cx={ css.iconBlue } />
                        <Text lineHeight="24" size="24" fontSize="14" cx={ css.dropCaption }>
                            {i18n.fileUpload.labelStart}
                        </Text>
                        <UploadFileToggler
                            onFilesAdded={ props.onUploadFiles }
                            render={ (props) => (
                                <RichTextView>
                                    <LinkButton caption={ i18n.fileUpload.browse } { ...props } cx={ css.link } />
                                </RichTextView>
                            ) }
                            accept={ props.accept }
                            single={ props.single }
                        />
                    </FlexRow>
                    {getInfoText}
                </div>
            </div>
        );
    };

    return <UuiDropSpot render={ renderAttachmentArea } onFilesDropped={ props.onUploadFiles } />;
}
