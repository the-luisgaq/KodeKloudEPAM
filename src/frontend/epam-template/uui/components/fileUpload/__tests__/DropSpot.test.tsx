import React from 'react';
import { DropSpot } from '../DropSpot';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';

describe('DropSpot', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<DropSpot onUploadFiles={ jest.fn } />);
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', async () => {
        const tree = await renderSnapshotWithContextAsync(<DropSpot onUploadFiles={ jest.fn } infoText="Test info" />);
        expect(tree).toMatchSnapshot();
    });
});
