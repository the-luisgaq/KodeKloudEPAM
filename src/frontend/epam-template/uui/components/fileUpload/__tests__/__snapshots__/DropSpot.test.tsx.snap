// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropSpot should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="dropArea"
    >
      <div
        class="root uui-flex-row uui-size-24"
      >
        <div
          class="container uui-icon uui-enabled iconBlue"
        >
          <svg
            class=""
          />
        </div>
        <div
          class="root uui-text uui-size-24 lineHeight fontSize uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography-inline dropCaption container"
          style="--uui-text-font-size: 14px; --uui-text-line-height: 24px;"
        >
          Drop files to attach or
        </div>
        <div>
          <input
            class="fileInput"
            multiple=""
            type="file"
          />
          <div
            class="uui-typography uui-typography-size-14"
          >
            <button
              class="uui-button-box uui-enabled -clickable uui-link_button root uui-size-36 uui-no-left-icon uui-no-right-icon uui-color-primary link"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption uui-link-button-weight-semibold"
              >
                browse
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DropSpot should be rendered correctly with props 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="dropArea"
    >
      <div
        class="root uui-flex-row uui-size-24"
      >
        <div
          class="container uui-icon uui-enabled iconBlue"
        >
          <svg
            class=""
          />
        </div>
        <div
          class="root uui-text uui-size-24 lineHeight fontSize uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography-inline dropCaption container"
          style="--uui-text-font-size: 14px; --uui-text-line-height: 24px;"
        >
          Drop files to attach or
        </div>
        <div>
          <input
            class="fileInput"
            multiple=""
            type="file"
          />
          <div
            class="uui-typography uui-typography-size-14"
          >
            <button
              class="uui-button-box uui-enabled -clickable uui-link_button root uui-size-36 uui-no-left-icon uui-no-right-icon uui-color-primary link"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption uui-link-button-weight-semibold"
              >
                browse
              </div>
            </button>
          </div>
        </div>
      </div>
      <div
        class="root uui-text uui-size-24 lineHeight fontSize uui-color-tertiary uui-font-weight-400 uui-font-style-normal uui-typography-inline container"
        style="--uui-text-font-size: 14px; --uui-text-line-height: 24px;"
      >
        Test info
      </div>
    </div>
  </div>
</DocumentFragment>
`;
